rcl/src/rcl/stdatomic_helper/win32/stdatomic.h

```diff
@@ -0,0 +1,361 @@
+      default: \
+        break; \
+    } \
+  } while (0); \
+  __pragma(warning(pop))
+
+#define rcl_win32_atomic_fetch_and(object, out, operand) \
+  __pragma(warning(push)) \
+  __pragma(warning(disable: 4244)) \
+  do { \
+    switch (sizeof(object)) { \
+      case sizeof(uint64_t): \
+        out = _InterlockedAnd64((LONGLONG *) object, operand); \
+        break; \
+      case sizeof(uint32_t): \
+        out = _InterlockedAnd((LONG *) object, operand); \
+        break; \
+      case sizeof(uint16_t): \
+        out = _InterlockedAnd16((SHORT *) object, operand); \
+        break; \
+      default: \
+        break; \
+    } \
+  } while (0); \
+  __pragma(warning(pop))
+
+#define rcl_win32_atomic_fetch_or(object, out, operand) \
+  __pragma(warning(push)) \
+  __pragma(warning(disable: 4244)) \
+  do { \
+    switch (sizeof(object)) { \
+      case sizeof(uint64_t): \
+        out = _InterlockedOr64((LONGLONG *) object, operand); \
+        break; \
+      case sizeof(uint32_t): \
+        out = _InterlockedOr((LONG *) object, operand); \
+        break; \
+      case sizeof(uint16_t): \
+        out = _InterlockedOr16((SHORT *) object, operand); \
+        break; \
+      default: \
+        break; \
+    } \
+  } while (0); \
+  __pragma(warning(pop))
+
+#define rcl_win32_atomic_fetch_sub(object, out, operand) \
+  rcl_win32_atomic_fetch_add(object, out, -(operand))
+
+#define rcl_win32_atomic_fetch_xor(object, out, operand) \
+  __pragma(warning(push)) \
+  __pragma(warning(disable: 4244)) \
+  do { \
+    switch (sizeof(object)) { \
+      case sizeof(uint64_t): \
+        out = _InterlockedXor64((LONGLONG *) object, operand); \
+        break; \
+      case sizeof(uint32_t): \
+        out = _InterlockedXor((LONG *) object, operand); \
+        break; \
+      case sizeof(uint16_t): \
+        out = _InterlockedXor16((SHORT *) object, operand); \
+        break; \
+      default: \
+        break; \
+    } \
+  } while (0); \
+  __pragma(warning(pop))
+
+#define rcl_win32_atomic_load(object, out) \
+  __pragma(warning(push)) \
+  __pragma(warning(disable: 4244)) \
+  do { \
+    switch (sizeof(object)) { \
+      case sizeof(uint64_t): \
+        out = _InterlockedExchangeAdd64((LONGLONG *) object, 0); \
+        break; \
+      case sizeof(uint32_t): \
+        out = _InterlockedExchangeAdd((LONG *) object, 0); \
+        break; \
+      case sizeof(uint16_t): \
+        out = _InterlockedExchangeAdd16((SHORT *) object, 0); \
+        break; \
+      default: \
+        break; \
+    } \
+  } while (0); \
+  __pragma(warning(pop))
+
+#define rcl_win32_atomic_store(object, desired) \
+  do { \
+    MemoryBarrier(); \
+    (object)->__val = (desired); \
+    MemoryBarrier(); \
+  } while (0)
+
+/*
+ * 7.17.8 Atomic flag type and operations. (disabled for now)
+ */
+
+// typedef atomic_bool     atomic_flag;
+
+// #define ATOMIC_FLAG_INIT    ATOMIC_VAR_INIT(0)
+
+// #define atomic_flag_clear_explicit(object, order) \
```

Comment:
Well I sort of wrote this (I used the FreeBSD header as a template, but I replaced the guts). They are commented out because I didn't need them :smile:. I can remove them completely if you like, but I figured I might uncomment them and update them for Windows at some point so I didn't yet.